framework/web/Request.php

```diff
@@ -196,6 +243,34 @@ public function resolve()
     }
 
     /**
+     * Filters headers according to the trusted host config.
+     * @param array $headers
+     * @since 2.0.13
+     * @return array filtered headers.
+     */
+    protected function filterHeaders(array $headers)
+    {
+        $host = $this->getUserHost();
```

Comment:
this adds an unnecessary overhead of making DNS query to every request. In most cases this is not needed so it should only be performed if hostname is really required.